kp_test.go

```diff
@@ -1070,3 +1071,77 @@ func TestDo_ConnectionError_HasCorrelationID(t *testing.T) {
 	urlErr := err.(*URLError)
 	assert.NotEmpty(t, urlErr.CorrelationID)
 }
+
+func TestDo_KPErrorResponseWithReasons_IsErrorStruct(t *testing.T) {
+	defer gock.Off()
+
+	errorWithReasons := []byte(`{
+		"metadata": {
+			"collectionType": "application/vnd.ibm.kms.error+json",
+			"collectionTotal": 1
+		},
+		"resources": [
+			{
+				"errorMsg": "Conflict: Action could not be performed on key. Please see reasons for more details.",
+				"reasons": [
+					{
+						"code": "KEY_ROTATION_NOT_PERMITTED",
+						"message": "This root key has been rotated within the last hour. Only one 'rotate' action per hour is permitted",
+						"status": 409,
+						"moreInfo": "https://cloud.ibm.com/apidocs/key-protect"
+					}
+				]
+			}
+		]
+	}`)
```

Comment:
Does `go fmt` make it indent like this...?

Reply:
no, I copied the response from Postman.